src/main/java/kr/syeyoung/dungeonsguide/dungeon/doorfinder/CatacombDataProvider.java

```diff
@@ -18,154 +18,119 @@
-                    BlockPos test = pos.add(vector2d.x * i, 0, vector2d.y * i);
-                    if (w.getChunkFromBlockCoords(test).getBlock(test) == Blocks.iron_bars) {
-                        return pos.add(vector2d.x * (i + 2), -2, vector2d.y * (i+2));
-                    }
-                }
-            }
+        if (!armorStand.isEmpty()) {
+            return getVector2d(w, armorStand, directions);
         }
         return null;
     }
 
-    @Override
-    public Vector2d findDoorOffset(World w, String dungeonName) {
-        Collection<EntityArmorStand> armorStand = w.getEntities(EntityArmorStand.class, new Predicate<EntityArmorStand>() {
-            @Override
-            public boolean apply(EntityArmorStand input) {
-                return input.getName().equals("§bMort");
-            }
-        });
-
-        if (armorStand.size() != 0) {
-            EntityArmorStand mort = armorStand.iterator().next();
-            BlockPos pos = mort.getPosition();
-            pos = pos.add(0, 3, 0);
-            for (int i = 0; i < 5; i++) {
-                for (Vector2d vector2d:directions) {
-                    BlockPos test = pos.add(vector2d.x * i, 0, vector2d.y * i);
-                    if (w.getChunkFromBlockCoords(test).getBlock(test) == Blocks.iron_bars) {
-                        return vector2d;
-                    }
+    @Nullable
+    static Vector2d getVector2d(World w, Collection<EntityArmorStand> armorStand, Set<Vector2d> directions) {
+        EntityArmorStand mort = armorStand.iterator().next();
+        BlockPos pos = mort.getPosition();
+        pos = pos.add(0, 3, 0);
+        for (int i = 0; i < 5; i++) {
+            for (Vector2d vector2d: directions) {
+                BlockPos test = pos.add(vector2d.x * i, 0, vector2d.y * i);
+                if (w.getChunkFromBlockCoords(test).getBlock(test) == Blocks.iron_bars) {
+                    return vector2d;
                 }
             }
         }
         return null;
     }
-    /*
-     *
-    * */
 
     @Override
     public BossfightProcessor createBossfightProcessor(World w, String dungeonName) {
         String floor = dungeonName.substring(14).trim();
         DungeonsGuide.sendDebugChat(new ChatComponentText("Floor: "+floor+ " Building bossfight processor"));
-        if (floor.equals("F1")) {
-            return new BossfightProcessorBonzo();
-        } else if (floor.equals("F2")) {
-            return new BossfightProcessorScarf();
-        } else if (floor.equals("F3")) {
-            return new BossfightProcessorProf();
-        } else if (floor.equals("F4")) {
-            return new BossfightProcessorThorn();
-        } else if (floor.equals("F5")) {
-            return new BossfightProcessorLivid(false);
-        } else if (floor.equals("F6")) {
-            return new BossfightProcessorSadan();
-        } else if (floor.equals("F7")) {
-            return new BossfightProcessorNecron();
+        switch (floor) {
```

Comment:
uhhhhhhhhhhhhhhhhhhhhhhhhhhhhh ok (i'm very against switch cases unless code looks cleaner and shorter with it)